Detectors/CPV/reconstruction/src/Clusterer.cxx

```diff
@@ -0,0 +1,419 @@
+            break;
+          case 2: // too far from each other
+          default:
+            break;
+        } // switch
+      }
+    } // loop over cluster
+  }   // energy theshold
+}
+//__________________________________________________________________________
+void Clusterer::makeUnfoldings(gsl::span<const Digit> digits)
+{
+  //Split cluster if several local maxima are found
+
+  std::vector<int> maxAt(o2::cpv::CPVSimParams::Instance().mNLMMax); // NLMMax:Maximal number of local maxima
+
+  int numberOfNotUnfolded = mClusters.size();
+
+  for (int i = 0; i < numberOfNotUnfolded; i++) { //can not use iterator here as list can expand
+    FullCluster& clu = mClusters[i];
+    if (clu.getNExMax() > -1) { //already unfolded
+      continue;
+    }
+    char nMultipl = clu.getMultiplicity();
+    char nMax = clu.getNumberOfLocalMax(maxAt);
+    if (nMax > 1) {
+      unfoldOneCluster(clu, nMax, maxAt, digits);
+      clu.setEnergy(0); // will be skipped later
+    } else {
+      clu.setNExMax(nMax); // Only one local maximum
+    }
+  }
+}
+//____________________________________________________________________________
+void Clusterer::unfoldOneCluster(FullCluster& iniClu, char nMax, gsl::span<int> digitId, gsl::span<const Digit> digits)
+{
+  // Performs the unfolding of a cluster with nMax overlapping showers
+  // Parameters: iniClu cluster to be unfolded
+  //             nMax number of local maxima found (this is the number of new clusters)
+  //             digitId: index of digits, corresponding to local maxima
+  //             maxAtEnergy: energies of digits, corresponding to local maxima
+
+  // Take initial cluster and calculate local coordinates of digits
+  // To avoid multiple re-calculation of same parameters
+  char mult = iniClu.getMultiplicity();
+  std::vector<std::vector<float>> eInClusters(mult, std::vector<float>(nMax));
+  std::vector<std::vector<float>> fij(mult, std::vector<float>(nMax));
+
+  const std::vector<FullCluster::CluElement>* cluElist = iniClu.getElementList();
+
+  // Coordinates of centers of clusters
+  std::vector<float> xMax(nMax);
+  std::vector<float> zMax(nMax);
+  std::vector<float> eMax(nMax);
+  std::vector<float> deNew(nMax);
+
+  //transient variables
+  std::vector<float> a(nMax);
+  std::vector<float> b(nMax);
+  std::vector<float> c(nMax);
+
+  for (int iclu = 0; iclu < nMax; iclu++) {
+    xMax[iclu] = (*cluElist)[digitId[iclu]].localX;
+    zMax[iclu] = (*cluElist)[digitId[iclu]].localZ;
+    eMax[iclu] = 2. * (*cluElist)[digitId[iclu]].energy;
+  }
+
+  std::vector<float> prop(nMax); // proportion of clusters in the current digit
+
+  // Try to decompose cluster to contributions
+  int nIterations = 0;
+  bool insuficientAccuracy = true;
+  while (insuficientAccuracy && nIterations < o2::cpv::CPVSimParams::Instance().mNMaxIterations) {
+    insuficientAccuracy = false; // will be true if at least one parameter changed too much
+    a.clear();
```

Comment:
here you clear a, b and c, but I don't see where do you resize them before accessing as ``a[..]`` etc. Again, switch to fixed arrays or even array of arrays (since they all seem to have fixed size nMax), then you can reset all of them in one go via memset.